cli/credentials/credentials.go

```diff
@@ -0,0 +1,123 @@
+package credentials
+
+import (
+	"fmt"
+	"github.com/rs/zerolog/log"
+	"github.com/saucelabs/saucectl/internal/yaml"
+	"golang.org/x/net/context"
+	yamlbase "gopkg.in/yaml.v2"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+)
+
+// Credentials contains a set of Username + AccessKey for SauceLabs.
+type Credentials struct {
+	Username  string `yaml:"username"`
+	AccessKey string `yaml:"accessKey"`
+	Source    string
+}
+
+// GetCredentials returns the currently configured credentials (env is prioritary vs. file).
+func GetCredentials() *Credentials {
+	if envCredentials := GetCredentialsFromEnv(); envCredentials != nil {
+		return envCredentials
+	}
+	return GetCredentialsFromFile()
+}
+
+// GetCredentialsFromEnv reads the credentials from the user environment.
+func GetCredentialsFromEnv() *Credentials {
+	creds := Credentials{
+		Username:  os.Getenv("SAUCE_USERNAME"),
+		AccessKey: os.Getenv("SAUCE_ACCESS_KEY"),
+		Source: "Environment",
+	}
+	if creds.IsEmpty() || !creds.IsValid() {
+		return nil
+	}
+	return &creds
+}
+
+// GetCredentialsFromFile reads the credentials from the user credentials file.
+func GetCredentialsFromFile() *Credentials {
+	var c *Credentials
+
+	if _, err := os.Stat(getCredentialsFolderPath()); err != nil {
+		log.Debug().Msgf("%s: config folder does not exists: %v", getCredentialsFolderPath(), err)
+		return nil
+	}
+
+	yamlFile, err := ioutil.ReadFile(getCredentialsFilePath())
+	if err != nil {
+		log.Info().Msgf("failed to read credentials: %v", err)
+		return nil
+	}
+
+	if err = yamlbase.Unmarshal(yamlFile, &c); err != nil {
+		log.Info().Msgf("failed to parse credentials: %v", err)
+		return nil
+	}
+	c.Source = getCredentialsFilePath()
+	if c.IsEmpty() || !c.IsValid() {
+		return nil
+	}
+	return c
+}
+
+func getCredentialsFolderPath() string {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		if "windows" == runtime.GOOS {
```

Comment:
That should be unnecessary. Please don't use SystemRoot.
If the home folder cannot be determined, feel free to just fail the operation.